BGL/include/CGAL/boost/graph/helpers.h

```diff
@@ -355,29 +356,207 @@ bool is_valid_face_descriptor( typename boost::graph_traits<FaceGraph>::face_des
   return true;
 }
 
-
-template <typename FaceGraph>
-bool is_valid_polygon_mesh(const FaceGraph& g)
+/*!
+  \ingroup PkgBGLHelperFct
+ * \brief is_valid_polygon_mesh checks the integrity of a `PolygonMesh`.
+ * \param g the `PolygonMesh` to test.
+ * \param verb : if `true`, the details of the check will be written in the standard output.
+ * \return `true` if the `PolygonMesh` is valid, `false` otherwise.
+ */
+template <typename PolygonMesh>
+bool is_valid_polygon_mesh(const PolygonMesh& g, bool verb = false)
 {
-  typedef typename boost::graph_traits<FaceGraph>::halfedge_descriptor halfedge_descriptor;
-  typedef typename boost::graph_traits<FaceGraph>::vertex_descriptor   vertex_descriptor;
-  typedef typename boost::graph_traits<FaceGraph>::face_descriptor     face_descriptor;
-  BOOST_FOREACH(vertex_descriptor v, vertices(g)){
-    if(! is_valid_vertex_descriptor(v,g)){
-      return false;
+  typedef typename boost::graph_traits<PolygonMesh>::halfedge_descriptor   halfedge_descriptor;
+  typedef typename boost::graph_traits<PolygonMesh>::vertex_descriptor     vertex_descriptor;
+  typedef typename boost::graph_traits<PolygonMesh>::face_descriptor       face_descriptor;
+  typedef typename boost::graph_traits<PolygonMesh>::vertex_iterator       vertex_const_iterator;
+  typedef typename boost::graph_traits<PolygonMesh>::halfedge_iterator     halfedge_const_iterator;
+  typedef typename boost::graph_traits<PolygonMesh>::face_iterator         face_const_iterator;
+  typedef typename boost::graph_traits<PolygonMesh>::vertices_size_type    vertex_size_type;
+  typedef typename boost::graph_traits<PolygonMesh>::halfedges_size_type   halfedges_size_type;
+  typedef typename boost::graph_traits<PolygonMesh>::faces_size_type       faces_size_type;
+  Verbose_ostream verr(verb);
+  int num_v(std::distance(boost::begin(vertices(g)), boost::end(vertices(g)))),
```

Comment:
`int` will trigger a conversion warning, use `std::size_t` for example. The real type is `std::iterator_traits<Iterator>::difference_type`